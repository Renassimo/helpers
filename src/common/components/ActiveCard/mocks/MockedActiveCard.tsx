const MockedActiveCard = jest.fn(
  ({ chosen, imageUrl, imageAlt, link, children }) => (
    <div>
      MockedActiveCard - {chosen && 'chosen'} - {imageUrl} - {imageAlt} - {link}{' '}
      - {children}
    </div>
  )
);

export default MockedActiveCard;
