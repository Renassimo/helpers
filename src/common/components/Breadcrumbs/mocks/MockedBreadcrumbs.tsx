import { BreadcrumbsItem } from '@/common/types/props';

const MockedBreadcrumbs = jest.fn(({ data }: { data: BreadcrumbsItem[] }) => {
  return (
    <div>
      {data.map((item) => (
        <a href={item.href} key={item.title}>
          {item.title} - {item.current ?? 'Current'}
        </a>
      ))}
    </div>
  );
});

export default MockedBreadcrumbs;
