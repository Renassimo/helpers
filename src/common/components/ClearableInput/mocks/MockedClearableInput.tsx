const MockedClearableInput = jest.fn(
  ({ value, label, type, originalValue }) => {
    return (
      <div>
        MockedClearableInput - {value} - {label} - {type} - {originalValue}
      </div>
    );
  }
);

export default MockedClearableInput;
