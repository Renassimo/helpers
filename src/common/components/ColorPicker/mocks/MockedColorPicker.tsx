const MockedColorPicker = jest.fn(({ name, value, label, onChange, error }) => {
  return (
    <div>
      {name} - {value} - {label} - {onChange && 'onChange'} - {error}
    </div>
  );
});

export default MockedColorPicker;
