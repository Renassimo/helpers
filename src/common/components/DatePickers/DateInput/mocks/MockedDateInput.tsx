const MockedDateInput = jest.fn(
  ({ value, placeholder, label, ariaLabel, originalValue }) => (
    <div>
      MockedDateInput - {value} - {placeholder} - {label} - {ariaLabel} -{' '}
      {originalValue}
    </div>
  )
);

export default MockedDateInput;
