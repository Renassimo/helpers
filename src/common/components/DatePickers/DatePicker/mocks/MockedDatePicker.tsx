const MockedDatePicker = jest.fn(({ dayValue, staticPicker, openTo }) => (
  <div>
    MockedDatePicker - {dayValue} - {staticPicker && 'static'} - {openTo}
  </div>
));

export default MockedDatePicker;
