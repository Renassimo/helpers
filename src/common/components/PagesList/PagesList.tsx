import Link from 'next/link';

import Box from '@mui/material/Box';
import Grid from '@mui/material/Grid';
import Paper from '@mui/material/Paper';
import Typography from '@mui/material/Typography';

import { PageInfo } from '@/auth/types';

const PagesList = ({ pages }: { pages: PageInfo[] }) => {
  return (
    <Grid
      container
      justifyContent="center"
      alignItems="center"
      spacing={2}
      mt={5}
    >
      {pages?.map(({ title, path, onClick }) => (
        <Grid item key={title} xs={10} sm={6} md={4} lg={3} xl={3}>
          <Link
            href={path}
            onClick={(event) => {
              if (onClick) {
                event.preventDefault();
                onClick();
              }
            }}
          >
            <Paper>
              <Box p={1}>
                <Typography component="h2" variant="h6" textAlign="center">
                  {title}
                </Typography>
              </Box>
            </Paper>
          </Link>
        </Grid>
      ))}
    </Grid>
  );
};

export default PagesList;
