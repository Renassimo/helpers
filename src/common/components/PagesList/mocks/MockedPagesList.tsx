import { PageInfo } from '@/auth/types';

const MockedPagesList = jest.fn(({ pages }: { pages: PageInfo[] }) => {
  return (
    <div>
      {pages.map((page) => (
        <a href={page.path} key={page.title}>
          {page.title}
        </a>
      ))}
    </div>
  );
});

export default MockedPagesList;
