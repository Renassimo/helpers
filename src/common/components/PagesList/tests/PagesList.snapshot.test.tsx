import renderWithTheme from '@/common/tests/helpers/renderWithTheme';

import PagesList from '../PagesList';
import { mockedPageInfos } from '@/auth/types/mocks';

describe('PagesList snapshot', () => {
  test('renders successfully', () => {
    // Arange
    // Act
    const { baseElement } = renderWithTheme(
      <PagesList pages={mockedPageInfos} />
    );
    // Assert
    expect(baseElement).toMatchSnapshot();
  });
});
