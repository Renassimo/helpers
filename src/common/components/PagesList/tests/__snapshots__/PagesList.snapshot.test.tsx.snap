// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PagesList snapshot renders successfully 1`] = `
<body>
  <div>
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-1xsir0u-MuiGrid-root"
    >
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-10 MuiGrid-grid-sm-6 MuiGrid-grid-md-4 MuiGrid-grid-lg-3 MuiGrid-grid-xl-3 css-4031bd-MuiGrid-root"
      >
        <a
          href="5Book"
        >
          <div
            class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-1ps6pg7-MuiPaper-root"
          >
            <div
              class="MuiBox-root css-hpgf8j"
            >
              <h2
                class="MuiTypography-root MuiTypography-h6 css-1kgxqm0-MuiTypography-root"
              >
                5book
              </h2>
            </div>
          </div>
        </a>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-10 MuiGrid-grid-sm-6 MuiGrid-grid-md-4 MuiGrid-grid-lg-3 MuiGrid-grid-xl-3 css-4031bd-MuiGrid-root"
      >
        <a
          href="spotting"
        >
          <div
            class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-1ps6pg7-MuiPaper-root"
          >
            <div
              class="MuiBox-root css-hpgf8j"
            >
              <h2
                class="MuiTypography-root MuiTypography-h6 css-1kgxqm0-MuiTypography-root"
              >
                Spotting
              </h2>
            </div>
          </div>
        </a>
      </div>
    </div>
  </div>
</body>
`;
